web/publicendpointresource.go

```diff
@@ -291,118 +291,35 @@ func restAddPort(w *rest.ResponseWriter, r *rest.Request, client *node.ControlCl
 	}
 
 	glog.V(0).Infof("Add PORT with %s %#v", r.URL.Path, r)
-	glog.V(0).Infof("request = %#v", request)
-
-	// Validate the port number
-	scrubbedPort := svc.ScrubPortString(request.PortName)
-	portParts := strings.Split(scrubbedPort, ":")
-	if len(portParts) <= 1 {
-		err := fmt.Errorf("Invalid port address. Port address be \":[PORT NUMBER]\" or \"[IP ADDRESS]:[PORT NUMBER]\"")
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
-	port, err := strconv.Atoi(portParts[1])
-	if err != nil {
-		err := fmt.Errorf("Port must be a number between 1 and 65536")
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
-
-	if port < 1 || port > 65535 {
-		err := fmt.Errorf("Port must be a number between 1 and 65536")
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
-
-	switch port {
-	case 5000:
-		fallthrough
-	case 8080:
-		err := fmt.Errorf("Port %d is reserved", port)
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
-
-	var services []svc.Service
-	var serviceRequest dao.ServiceRequest
-	if err := client.GetServices(serviceRequest, &services); err != nil {
-		err := fmt.Errorf("Could not get services: %v", err)
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
-
-	var service *svc.Service
-	for _, _service := range services {
-		if _service.ID == request.ServiceID {
-			service = &_service
-			break
-		}
-	}
 
-	if service == nil {
-		err := fmt.Errorf("Could not find service: %s", request.ServiceID)
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
+	facade := ctx.getFacade()
+	dataCtx := ctx.getDatastoreContext()
 
-	//checkout other ports for redundancy
-	for _, service := range services {
-		if service.Endpoints == nil {
-			continue
-		}
-
-		for _, endpoint := range service.Endpoints {
-			for _, epPort := range endpoint.PortList {
-				if scrubbedPort == epPort.PortAddr {
-					err := fmt.Errorf("Port %s already defined for service: %s", epPort.PortAddr, service.Name)
-					glog.Error(err)
-					restServerError(w, err)
-					return
-				}
-			}
-		}
-	}
-
-	// Check to make sure the port is available.  Don't allow adding a port if it's already being used.
-	err = checkPort("tcp", fmt.Sprintf("%s", scrubbedPort))
+	// Get the service for this service id.
+	service, err := facade.GetService(dataCtx, request.ServiceID)
```

Comment:
This is in errors like "glog.Errorf("Error adding port to service (%s): %v", service.Name, err)", which is displayed in a flare in the dialog.